packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextRepresentationDataFetcher.java

```diff
@@ -66,22 +64,20 @@ public DataFetcherResult<RepresentationMetadata> get(DataFetchingEnvironment env
 
         Map<String, Object> localContext = new HashMap<>(environment.getLocalContext());
         localContext.put(LocalContextConstants.REPRESENTATION_ID, representationId);
-
         // Search among the active representations first. They are already loaded in memory and include transient
         // representations.
         // @formatter:off
         var representationMetadata = this.editingContextEventProcessorRegistry.getEditingContextEventProcessors().stream()
                     .flatMap(editingContextEventProcessor -> editingContextEventProcessor.getRepresentationEventProcessors().stream())
                     .filter(editingContextEventProcessor -> editingContextEventProcessor.getRepresentation().getId().equals(representationId))
                     .map(IRepresentationEventProcessor::getRepresentation)
-                    .filter(ISemanticRepresentation.class::isInstance)
-                    .map(ISemanticRepresentation.class::cast)
-                    .map((ISemanticRepresentation representation) -> {
+                    .filter(IRepresentation.class::isInstance)
+                    .map(IRepresentation.class::cast)
+                    .map((IRepresentation representation) -> {
```

Comment:
`IRepresentationEventProcessor#getRepresentation` already returns an `IRepresentation`, you don't need to filter it, cast it nor type it after that.